Backend/controllers/auth/signup.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response } from "express";
+import { encrypt } from "@omar-sarfraz/caesar-cipher";
+import bcrypt from "bcrypt";
+
+import { UserType } from "../../routes/auth/auth";
+import { User } from "../../models/User";
+
+const signup = async (req: Request, res: Response) => {
+    const userData: UserType = req.body;
+
+    if (!userData.email || !userData.firstName || !userData.lastName || !userData.password)
+        return res.status(400).json({
+            message: "firstName, lastName, email and password is required!",
+            error: true,
+        });
+
+    try {
+        const key: number = parseInt(process.env.KEY || "");
+        if (!key) {
+            console.error("Key is required!");
+            process.exit(1);
+        }
+
+        const encryptedEmail = encrypt(key, userData.email);
+
+        const existingUser = await User.findOne({ where: { email: encryptedEmail } });
+        if (existingUser) {
+            return res
+                .status(400)
+                .json({ message: "User with this email already exists", error: true });
+        }
+
+        const encryptedFirstName = encrypt(key, userData.firstName);
+        const encryptedLastName = encrypt(key, userData.lastName);
+
+        const hash = await bcrypt.genSalt(key);
+        const encryptedPassword = await bcrypt.hash(userData.password, hash);
+
+        const user = await User.create({
+            firstName: encryptedFirstName,
```

Comment:
lets not encrypt first name and last name.

Reply:
Alright